docs/reference/content/bson/documents.md

```diff
@@ -0,0 +1,139 @@
++++
+date = "2015-03-19T14:27:51-04:00"
+title = "Documents"
+[menu.main]
+  parent = "BSON"
+  weight = 20
+  pre = "<i class='fa'></i>"
++++
+
+## Documents
+
+The scala driver includes two scala specific representations for BSON documents. Following convention from the scala collections library,
+there are immutable and mutable implementations of documents. The underlying implementations of the scala Document use the type safe 
+[`BsonDocument`]({{< coreapiref "org/bson/BsonDocument" >}}) class. The scala bson classes are available from the `org.mongodb.scala.bson` 
+namespace, which includes type aliases and companion objects.  In general this should suffice but for advanced use cases you may need to 
+use `org.bson` directly.
+
+{{% note %}}
+The scala `Document` classes implement `TraversableLike[(String, BsonValue)]` and the general API mirrors that of a `Map[String, BsonValue]`.
+However, unlike `Map` implementations of `TraversableLike` enables strict type safety as their is no variance in the value type.
+{{% /note %}}
+
+[`BsonValue`]({{< apiref "org.mongodb.scala.bson.BsonValue" >}}) is the type safe representation of a Bson type from the `org.bson` library, 
+it represents specific value types, the most commonly used value types are: 
+   
+| BSON type | Scala type                        |
+|-----------|-----------------------------------|
+| Document  | `org.mongodb.scala.bson.Document` |
+| Array     | `List`                            |
+| Date      | `Date` or `int` (ms since epoch)  |
+| Boolean   | `Boolean`                         |
+| Double    | `Double`                          |
+| Int32     | `Integer`                         |
+| Int64     | `Long`                            |
+| String    | `String`                          |
+| Binary    | `Array[Byte]`                     |
+| ObjectId  | `ObjectId`                        |
+| Null      | `None`                            |
+
+It is actually possible to change or extend these mappings this will be covered in detail below.
+
+There are two main `Document` classes:
+
+## Immutable Documents
+
+Like the Scala collections library the immutable class is the favoured class.  For convenience it is aliased to `org.mongodb.scala.Document`
+and `org.mongodb.scala.bson.Document` as well as being available from `org.mongodb.scala.bson.collection.immutable.Document`. Instances of 
+this type are guaranteed to be immutable for everyone. Such a collection will never change after it is created. Therefore, you can rely on 
+the fact that accessing the same collection value repeatedly at different points in time will always yield a collection with the same elements.
+
+```scala
+import org.mongodb.scala.bson._
+
+val doc1 = Document("AL" -> BsonString("Alabama"))
+val doc2 = doc1 + ("AK" -> BsonString("Alaska"))
+val doc3 = doc2 ++ Document("AR" -> BsonString("Arkansas"), "AZ" -> BsonString("Arizona"))
```

Comment:
I mean an example of how to add an array of BSON values, e.g.

   val docs4 = docs3 ++ [BsonInt32(1), BsonInt32(5)]